server/src/main/java/org/elasticsearch/rest/CompatibleHandlers.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.rest;
+
+import org.elasticsearch.common.logging.DeprecationLogger;
+import org.elasticsearch.common.xcontent.ToXContent;
+
+import java.util.function.Consumer;
+
+public class CompatibleHandlers {
+
+    /**
+     * TODO revisit when https://github.com/elastic/elasticsearch/issues/52370 is resolved
+     */
+    public static final String COMPATIBLE_HEADER = "Accept";
+    public static final String COMPATIBLE_PARAMS_KEY = "Compatible-With";
+    public static final String COMPATIBLE_VERSION = "7";
```

Comment:
Can we make this major version - 1 , so that when v9 it is auto incremented to 8 ?

Reply:
good idea, this one will always stay as it is part of generic code always used for compatible API